app/components/PokemonDetails.js

```diff
@@ -0,0 +1,205 @@
+import React from 'react';
+import { View, Image, StyleSheet, Text, ScrollView, AsyncStorage, } from "react-native";
+
+class PokemonDetails extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state ={ 
+            name: "",
+            image: null,
+            weight: "",
+            height: "",
+            moves: [],
+            types: [],
+            data: {},
+            show : this.props.visible
+        };
+        this.get = this.get.bind(this);
+    }
+    componentWillReceiveProps(props) {
+        this.setState({  name : props.name});
+    }
+
+    async get(key) {
+        try {
+            const value = await AsyncStorage.getItem(key).then((val) => {
+                return JSON.parse(val);
+            });
+            return value;
+        } catch (error) {
+            console.error(error);
+            return null;
+        }
+    
+    }
+
+    render() {
+
```

Comment:
When you have a render with a lot of logic its a good practice break down in small functions